### PR TITLE
#110 Enhance registration process to use the email queue

### LINKED ISSUE #110: Enhance registration process to use the email queue
As part of the solution, we will have an email service that will consume email requests of a queue and send emails.
We need to enhance the registration process to use the RabbitMQ interface (#105 ) to send requests to the email service via the email queue.

### PR DESCRIPTION
**Pull request by virtual4real**
### Requirements for making a pull request

Thank you for contributing to our project! 

Please fill out the template below to help the project maintainers review it as fast as possible and include your contribution to the project.

### What does it fix?

Closes https://github.com/code4romania/stam-acasa/issues/110

Adds EasyMQ to Identity project and sends a message to the MQ in the user registration process. The messsage contains the user info and it is received by the EmailService that creates and sends an email.

### How has it been tested?

The template for the registration email is very simple and needs formatting.

### CODE REVIEW

File: backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml.cs [hunk 1]
```diff
@@ -26,20 +27,23 @@ public class RegisterModel : PageModel
...
+        // private readonly IEmailSender _emailSender;
+        private readonly IQueueService _queue;
         private readonly IConfiguration _configuration;
 
         public RegisterModel(
             UserManager<ApplicationUser> userManager,
             SignInManager<ApplicationUser> signInManager,
             ILogger<RegisterModel> logger,
-            IEmailSender emailSender,
+            // IEmailSender emailSender,
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml.cs]: Let's remove commented code.

File: backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml.cs [hunk 2]
```diff
@@ -26,20 +27,23 @@ public class RegisterModel : PageModel
...
+            IQueueService queueService,
             IConfiguration configuration)
         {
             _userManager = userManager;
             _signInManager = signInManager;
             _logger = logger;
-            _emailSender = emailSender;
+            // _emailSender = emailSender;
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml.cs]: Let's remove commented code.

File: backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml.cs [hunk 3]
```diff
@@ -106,17 +124,18 @@ public async Task<IActionResult> OnPostAsync(string returnUrl = null)
...
-                                To = Input.Email,
-                                Subject = "Confirm your email",
-                                Content = $"Please confirm your account by <a href='{HtmlEncoder.Default.Encode(callbackUrl)}'>clicking here</a>.",
-                                FromName = _configuration.GetValue<string>("AdminFromName"),
-                                FromEmail = _configuration.GetValue<string>("AdminFromEmail")
-                            },
-                            default
-                        );
+                        await SendRegistrationEmail(user.UserName, callbackUrl);
+                        //await _emailSender.SendAsync(
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.IdentityServer/Areas/Identity/Pages/Account/Register.cshtml.cs]: Let's remove commented code.

File: backend/src/StamAcasa.IdentityServer/Startup.cs [hunk 4]
```diff
@@ -74,13 +74,20 @@ public void ConfigureServices(IServiceCollection services)
                     );
                     break;
             }
+
+            services.AddSingleton(RabbitHutch.CreateBus(string.Format("host={0}:{1};username={2};password={3}",
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.IdentityServer/Startup.cs:78]: Do we need this anymore, as we are using the IQueueService implementation?

File: backend/src/StamAcasa.IdentityServer/appsettings.json [hunk 5]
```diff
@@ -8,7 +8,7 @@
   },
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "ApplicationDbContextConnection": "Server=postgres;Port=5432;Database=IdentityServer;User Id=docker;Password=docker;"
+    "ApplicationDbContextConnection": "Data Source=IdentityServer.db"
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.IdentityServer/appsettings.json]: Any reason why you switched back to the in memory database?

**Comment by RaduCStefanescu** [comment]
Hey @virtual4real 
Many thanks for this PR!
I left some comments, if you can handle them we can merge this one.

File: backend/src/StamAcasa.IdentityServer/Startup.cs (hunk 4 above)
Comment (virtual4real) [reply backend/src/StamAcasa.IdentityServer/Startup.cs:78]: Yes, the QueueService constructor needs a parameter of type IBus so the bus has to be created here to be resolved when QueueService is instantiated.

File: backend/src/StamAcasa.IdentityServer/appsettings.json (hunk 5 above)
Comment (virtual4real) [reply backend/src/StamAcasa.IdentityServer/appsettings.json]: No,  I was just working locally with the in memory db

**Review by RaduCStefanescu — APPROVED** [review]

**Pull request merged**